kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/RecordEncryptionFilter.java

```diff
@@ -102,7 +102,7 @@ private CompletionStage<ProduceRequestData> maybeEncodeProduce(ProduceRequestDat
                 }).exceptionallyCompose(throwable -> {
                     log.atWarn().setMessage("failed to encrypt records, cause message: {}")
                             .addArgument(throwable.getMessage())
-                            .setCause(log.isDebugEnabled() ? throwable : null)
+                            .setCause(log.isDebugEnabled() ? throwable.getCause() : null)
```

Comment:
I don't think this is right. 

* We'd miss the detail of what the throwable actual is.  That might be important to problem resolution.
* What if throwable has no cause?  The logging line will then throw a NPE.

Reply:
> - We'd miss the detail of what the throwable actual is. That might be important to problem resolution.

That's a good point. I agree, it's not ideal.

> - What if throwable has no cause? The logging line will then throw a NPE.

If the throwable has no cause then getCause will return `null`, per [the docs](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Throwable.html#getCause()):

> _Returns the cause of this throwable or null if the cause is nonexistent or unknown. (The cause is the throwable that caused this throwable to get thrown.)_

In that case `setCause` will be passed the same value that it gets when DEBUG logging is not enabled, which works fine. The log line would still be printed, but with no stack trace. However, if throwable was somehow `null` itself, then yes, the log line would fail.